report.py

```diff
@@ -56,34 +35,79 @@ def get_total_page_count(url: str, headers: Dict[str, Union[str, int]] = {}):
   "Authorization": f"Bearer {ENV['ZOOM_TOKEN']}"
 }
 
+ZOOM_BASE_URL = ENV.get('ZOOM_BASE_URL', "")
+DEFAULT_SLEEP_TIME = ENV.get('DEFAULT_SLEEP_TIME', "10")
 
-def run_report(param_attribute, headers, json_attribute_name, page_token=False):
 
-    url = ENV[f'ZOOM_{param_attribute}_URL']
-    params = ENV[f'ZOOM_{param_attribute}_PARAMS']
+# Functions
+def get_total_page_count(url: str, headers: Dict[str, Union[str, int]] = {}, params: dict = {}):
+    # get the total page count
+    total_page_count = 0
+    response = requests.request("GET", f"{url}", headers=headers, params=params)
+    status_code = response.status_code
+    if status_code != 200:
+        logger.warning(f'Received irregular status code: {status_code}')
+        logger.info('No page at all!')
+    else:
+        try:
+            results = json.loads(response.text.encode('utf8'))
+            total_page_count = results['page_count']
+        except json.JSONDecodeError:
+            logger.warning('JSONDecodeError encountered')
+            logger.info('No page at all!')
+    return total_page_count
 
-    # construct param string
-    f = furl.furl('')
-    f.args = params
-    print(f.url)
 
-    # get total page count
-    total_page_count = get_total_page_count(f"{url}{f.url}", headers)
+def run_report(api_url: str, headers: dict, json_attribute_name: str,
+               default_params: dict = {}, page_token: bool = False, use_date: bool = False):
+    url = ZOOM_BASE_URL+api_url
+    params = default_params
+    params['page_size'] = 300
+    total_list = []
+    # TODO: Detect the date from the previous CSV
+    # Either loop for all dates or just one a single report
+    if use_date:
+        early_date = dateparser.parse(ENV.get('ZOOM_EARLIEST_FROM', '2020-03-01')).date()
+        # Only use the date as a paramter
+        # Loop until yesterday (this will go until now() -1)
+        for i in range((datetime.now().date() - early_date).days):
+            param_date = early_date + timedelta(days=i)
+            params["from"] = str(param_date)
+            params["to"] = str(param_date)
+            # Add this loop to the list
+            total_list.extend(zoom_loop(url, headers, json_attribute_name, dict(params), page_token))
+    else:
+        total_list.extend(zoom_loop(url, headers, json_attribute_name, dict(params), page_token))
+    # output csv file
+    total_df = pd.DataFrame(total_list)
+    total_df.index.name = "index_id"
+    output_file_name = f"total_{json_attribute_name}.csv"
+    # Remove any duplicate uuids in the record
+    logger.info(f"Initial dataframe size: {len(total_df)}")
+    if "uuid" in total_df:
+        total_df.drop_duplicates("uuid", inplace=True)
+        logger.info(f"Dataframe with duplicates removed: {len(total_df)}")
```

Comment:
Would be interesting to see the number difference with and without the duplicates.

Reply:
Yeah, I'm seeing usually only seeing a few (3-8 duplicates). We could probably display the duplicates and I'm not sure why they're created, maybe something for the future.